pkg/lib/_internal/dnf5daemon.ts

```diff
@@ -0,0 +1,309 @@
+            throw err;
+        }
+
+        subscription.remove();
+        // HACK: close the client so subscribe matches are actually dropped. https://github.com/cockpit-project/cockpit/issues/21905
+        client.close();
+
+        return data;
+    }
+
+    async install_missing_packages(data: MissingPackages, progress_cb?: InstallProgressCB): Promise<void> {
+        if (!data || data.missing_ids.length === 0)
+            return;
+
+        let last_info: number;
+        let last_progress = 0;
+        let last_name: string;
+        let total_packages: number;
+
+        function signal_emitted(_path: string, _iface: string, signal: string, args: unknown[]) {
+            switch (signal) {
+            // download_add_new(o session_object_path, s download_id, s description, x total_to_download)
+            case 'download_add_new': {
+                last_info = InstallProgressType.DOWNLOADING;
+                last_name = args[2] as string;
+                break;
+            }
+            // download_progress(o session_object_path, s download_id, x total_to_download, x downloaded)
+            case 'download_progress': {
+                last_info = InstallProgressType.DOWNLOADING;
+                break;
+            }
+            // download_end(o session_object_path, s download_id, u transfer_status, s message)
+            case 'download_end':
+                last_info = 0;
+                last_name = "";
+                break;
+            // transaction_before_begin(o session_object_path, t total)
+            case 'transaction_before_begin':
+                [, total_packages] = args as [string, number];
+                last_info = InstallProgressType.INSTALLING;
+                break;
+            // transaction_elem_progress(o session_object_path, s nevra, t processed, t total)
+            case 'transaction_elem_progress': {
+                let processed = 0;
+                [, last_name, processed,] = args as [string, string, number, number];
+                last_progress = processed / total_packages * 100;
+                break;
+            }
+            }
+
+            if (progress_cb)
+                progress_cb({
+                    cancel: null,
+                    info: last_info,
+                    package: last_name,
+                    percentage: last_progress,
+                    waiting: false,
+                });
+        }
+
+        const client = dbus_client();
+        const subscription = client.subscribe({}, signal_emitted);
+        let session: string | null = null;
+
+        try {
+            session = await open_session();
+            await call(session, "org.rpm.dnf.v0.rpm.Rpm", "install", [data.missing_names, {}]);
+            const [, resolve_result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]);
+
+            if (resolve_result !== 0)
+                throw new ResolveError(`Resolving install failed with result={resolve_result}`);
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22397-64d6a5e5-20250908-152858-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)